File: IaAdmin/service/client/product_client.py
import httpx
from config.settings import PRODUCT_SERVICE_URL


async def get_active_products(store_id: str, jwt_token: str) -> list:
    """GET /api/v1/products/all/active"""
    try:
        async with httpx.AsyncClient(timeout=8) as client:
            r = await client.get(
                f"{PRODUCT_SERVICE_URL}/api/v1/products/all/active",
                headers={"X-Store-Id": store_id, "Authorization": f"Bearer {jwt_token}"}
            )
            if r.status_code == 200:
                data = r.json()
                return data if isinstance(data, list) else data.get("data", [])
    except Exception as e:
        print(f"[product_client] error: {e}")
    return []

File: Cms/src/main/java/com/api/Cms/clients/ProductClient.java
package com.api.Cms.clients;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.client.WebClient;

import java.time.Duration;
import java.util.List;
import java.util.Map;

@Component
@RequiredArgsConstructor
public class ProductClient {

    private final WebClient productWebClient;

    public List<Map<String, Object>> getActiveProducts(String storeId) {
        try {
            return productWebClient.get()
                    .uri("/api/v1/products/all/active")
                    .header("X-Store-Id", storeId)
                    .retrieve()
                    .bodyToFlux(Map.class)
                    .cast(Map.class)
                    .map(m -> (Map<String, Object>) m)
                    .collectList()
                    .timeout(Duration.ofSeconds(5))
                    .block();
        } catch (Exception e) {
            return List.of();
        }
    }
}